File: src/main/java/org/example/tree/LevelOrderTraversal.java
package org.example.tree;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class LevelOrderTraversal {

    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);

        while (!q.isEmpty()) {
            int qLen = q.size();
            List<Integer> level = new ArrayList<>();

            for (int i=0; i < qLen; i++) {
                TreeNode node = q.remove();
                if (node != null) {
                    level.add(node.val);
                    q.add(node.left);
                    q.add(node.right);
                }
            }
            if (!level.isEmpty()) res.add(level);
        }

        return res;
    }

}

File: src/main/java/org/example/tree/LevelOrderTraversal.py
from typing import List, Optional

from org.example.tree.TreeNode import TreeNode


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root: return []
        q = [root]
        res = []

        while q:
            qLen = len(q)
            level = []
            for i in range(qLen):
                node = q.pop(0)
                if node:
                    level.append(node.val)
                    q.append(node.left)
                    q.append(node.right)
            if level:
                res.append(level)
        return res
